core/src/main/java/org/wso2/carbon/kernel/utils/Utils.java

```diff
@@ -156,4 +173,207 @@ public static void checkSecurity() {
             secMan.checkPermission(new ManagementPermission("control"));
         }
     }
+
+    /**
+     * This method converts a given YAML String to XML format
+     *
+     * @param yamlString YAML String that needs to be converted to XML format
+     * @return String in XML format
+     */
+    public static String convertYAMLToXML(String yamlString, String rootElement) {
+        String jsonString = convertYAMLToJSON(yamlString);
+        return convertJSONToXML(jsonString, rootElement);
+    }
+
+    /**
+     * This method converts a given XML String to YAML format
+     *
+     * @param xmlString XML String that needs to be converted to YAML format
+     * @return String in YAML format
+     */
+    public static String convertXMLToYAML(String xmlString, String rootElement) {
+        String jsonString = convertXMLToJSON(xmlString);
+        return convertJSONToYAML(jsonString, rootElement);
+    }
+
+    /**
+     * This method converts a given YAML String to JSON format
+     *
+     * @param yamlString YAML String that needs to be converted to JSON format
+     * @return String in JSON format
+     */
+    public static String convertYAMLToJSON(String yamlString) {
+        String jsonString;
```

Comment:
This is fixed in the following PR.

https://github.com/wso2/carbon-kernel/pull/1161